### PR TITLE
Suggestions for first-time build instructions

### PR DESCRIPTION
**Pull request by awmartin**
Going through the updated build instructions went fairly smoothly overall. There were only a few things I needed to adjust in the process, so hopefully these will be useful for other new contributors:

* Specified the JDK version required to build Processing and Processing.py
* Added instructions to build Processing
* Notes about the out-of-the-box folder structure
* Pedantic language changes and formatting

### CODE REVIEW

File: HOWTO-HACK.txt [hunk 1]
```diff
@@ -2,54 +2,74 @@ Instructions for folks who want to contribute to processing.py itself.
 
 Prerequisites:
 
-	git - http://git-scm.com/downloads
-	The latest JDK - http://www.oracle.com/technetwork/java/javase/downloads/index.html
-	Apache ant - http://ant.apache.org/bindownload.cgi
+    git - http://git-scm.com/downloads
+    JDK 8u144 - http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html
```
Comment (jdf) [inline HOWTO-HACK.txt]: I don't want to have to update this file every time Processing changes their build requirement. Did you find that the "latest" didn't work?

File: HOWTO-HACK.txt [hunk 2]
```diff
@@ -2,54 +2,74 @@ Instructions for folks who want to contribute to processing.py itself.
...
 
-If you think you're going to want to submit your own work to processing.py,
-you should start by creating a fork of processing.py on github. Let's say
+    processing-source/
+        |- processing.py/
+        |- processing/
+        |- processing-video/
+
+If you think you're going to want to submit your own work to Processing.py,
+you should start by creating a fork of Processing.py on GitHub. Let's say
```
Comment (jdf) [inline HOWTO-HACK.txt]: `processing.py` not `Processing.py`

File: HOWTO-HACK.txt [hunk 3]
```diff
@@ -2,54 +2,74 @@ Instructions for folks who want to contribute to processing.py itself.
...
 
   
 You'll need source for processing and processing-video.
 
-  $ git clone --depth=1 https://github.com/processing/processing.git
-  $ git clone --depth=1 https://github.com/processing/processing-video.git
+    $ git clone --depth=1 https://github.com/processing/processing.git
+    $ git clone --depth=1 https://github.com/processing/processing-video.git
+
+Build processing core from source (according to the Processing build instructions):
```
Comment (jdf) [inline HOWTO-HACK.txt]: There's no reason to do this; the processing.py build should do it for you. Did it not?

File: HOWTO-HACK.txt (hunk 1 above)
Comment (awmartin) [reply HOWTO-HACK.txt]: Yeah, I couldn't get the JDK 9 line to work, the new line that that page references. It failed like this during the build:

ant_build_error_jdk

I assume it's non-trivial for the processing team to upgrade to the new major line of the JDK, being tied to JDK 8, or more specifically, 8u144, declared by the error message above and the "latest" release of JDK 8.

I don't think the build requirement, if for JDK 8, is much of a moving target, but perhaps it's sufficient to say "latest of the JDK 8?"

File: HOWTO-HACK.txt (hunk 2 above)
Comment (awmartin) [reply HOWTO-HACK.txt]: Will do!

File: HOWTO-HACK.txt (hunk 3 above)
Comment (awmartin) [reply HOWTO-HACK.txt]: It did! I tried again on a clean slate, and it all built in one go. Will remove.

File: HOWTO-HACK.txt (hunk 1 above)
Comment (jdf) [reply HOWTO-HACK.txt]: That would be great; thanks.

File: HOWTO-HACK.txt [hunk 4]
```diff
@@ -2,54 +2,74 @@ Instructions for folks who want to contribute to processing.py itself.
...
 Prerequisites:
 
-	git - http://git-scm.com/downloads
-	The latest JDK - http://www.oracle.com/technetwork/java/javase/downloads/index.html
-	Apache ant - http://ant.apache.org/bindownload.cgi
+    git - http://git-scm.com/downloads
+    JDK 8u144 - http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html
+    Apache ant - http://ant.apache.org/bindownload.cgi
+
+JDK 8u144 is required until such time as Processing is built with JDK 9 or later.
```
Comment (jdf) [inline HOWTO-HACK.txt]: This line is brittle; their specific "best version" changes frequently with the JDK release.

**Comment by awmartin** [comment]
Sorry. Github won't let me update the source branch of this PR, and I don't want to diverge from your master.

**Pull request closed without merging**